### DISCUSSION TITLE
Concerns vis a vis PIPx

### DISCUSSION THREAD

**Discussion by TamanBachani** [smartcontractkit/full-blockchain-solidity-course-py#25] [Q&A] [answered]
I have been trying to debug this for hours now. I installed pipx yet it doesn't recognize it as a command.

pipx : The term 'pipx' is not recognized as the name of a cmdlet, function, script file, or   
operable program. Check the spelling of the name, or if a path was included, verify that the  
path is correct and try again.
At line:1 char:1
+ pipx install eth-brownie
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (pipx:String) [], CommandNotFoundException      
    + FullyQualifiedErrorId : CommandNotFoundException

**Comment by LeCauchy** [comment]
Did you restart you terminal or your computer before trying to use it ? If not maybe you should.

**Comment by PatrickAlphaC** [comment, marked as answer]
Can you add the full output? I _think_ this is a pretty common error, but I need to see the rest of the error.

**Reply by krampus4570** [reply]
I was having the same issue- that was all that was displayed for the error. Apparently there are some problems around Pipx setting the path properly and reporting around the error as well. 

I finally got it to run by manually setting a path variable (https://www.computerhope.com/issues/ch000549.htm#windows10) to the folder containing pipx.exe- C:\Users\user\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts.

However once I was able to run Pipx, it kept hanging up while installing Brownie. I eventually just installed using only Pip from the Brownie docs though I'm not sure if there are any downsides to this approach yet.

**Reply by PatrickAlphaC** [reply]
The downsides are if you look to use different packages than what brownie uses, brownie will stop working.

However, using the same web3 tools brownie does right now are stable versions anyways, so you'll be fine. 

Could you link whatever resource helped your resolve this error?

**Reply by PatrickAlphaC** [reply]
Marking this as the answer for now, please follow up if you have more information

**Reply by krampus4570** [reply]
I dug through several resources, but this one was the most helpful: https://githubmemory.com/repo/pipxproject/pipx/issues/659

The only issue I have with my current setup is any time I use Brownie I get "INFO: Could not find files for the given pattern(s)." and there is a ~5 second delay before everything runs as it should. I might look into resolving that later, but my current setup got me through lesson 5 which was my goal for the day.

**Reply by PatrickAlphaC** [reply]
Boom!! HUGE Congrats!!
